### PR TITLE
WM-2646: Use WDS query to fetch rows in batches

### PR DESCRIPTION
**Pull request by cjllanwarne**

### CODE REVIEW

File: service/src/main/java/bio/terra/cbas/dependencies/wds/WdsService.java
Comment (salonishah11) [inline service/src/main/java/bio/terra/cbas/dependencies/wds/WdsService.java:108]: When this new filter was released, AJ team had mentioned that it is possible some of WDS apps in prod might not have been updated to have the new search filter for querying records as all the apps might not be updated. But it has been a while since this new version was released. Do you think it would be worth checking with AJ team to see if all existing WDS apps might have the new search filter?

Comment (salonishah11) [reply service/src/main/java/bio/terra/cbas/dependencies/wds/WdsService.java:108]: They had also mentioned that there is a `capabilities` API that can be used to figure out which features are available in a particular WDS version.

Comment (mspector) [reply service/src/main/java/bio/terra/cbas/dependencies/wds/WdsService.java:108]: +1... if we can determine that the WDS apps have all been updated in prod, this LGTM, otherwise I wonder if we should build in a call to this `capabilities` API to decide which call to make

File: service/src/main/java/bio/terra/cbas/dependencies/wds/WdsService.java
Comment (davidangb) [inline service/src/main/java/bio/terra/cbas/dependencies/wds/WdsService.java:97]: Ideally you'd also check here that the value of `search.filter.ids` is true, instead of just checking that the key exists.

File: service/src/main/java/bio/terra/cbas/dependencies/wds/WdsService.java
Comment (davidangb) [inline service/src/main/java/bio/terra/cbas/dependencies/wds/WdsService.java:110]: the capabilities api is a very lightweight implementation inside WDS, so it's no big deal … but conceivably you could check and cache the capabilities response for some amount of time. WDS's response will only change infrequently: when we perform app updates, and then only if something significant has changed in the app. I wouldn't block this PR on any kind of caching though.

File: service/src/main/java/bio/terra/cbas/dependencies/wds/WdsClient.java
Comment (mspector) [inline service/src/main/java/bio/terra/cbas/dependencies/wds/WdsClient.java:57]: 🎉

**Review by mspector — APPROVED** [review]
I would want to see David's comment addressed about `search.filter.ids`, but I don't have any other comments - LGTM

File: service/src/main/java/bio/terra/cbas/dependencies/wds/WdsService.java
Comment (salonishah11) [reply service/src/main/java/bio/terra/cbas/dependencies/wds/WdsService.java:110]: Yes we had discussed this briefly and I think we would want to cache this if its start affecting the performance or as part of future slice whichever happens first. But it is good to know that is a very lightweight implementation.

File: service/src/main/java/bio/terra/cbas/dependencies/wds/WdsService.java
Comment (salonishah11) [reply service/src/main/java/bio/terra/cbas/dependencies/wds/WdsService.java]: Added this check in [this commit](https://github.com/DataBiosphere/cbas/pull/289/commits/09962fe41d91cfbcd1bc5a66467f1acb251d6ab6)

**Review by kpierre13 — APPROVED** [review]

**Pull request merged**
